pkg/deploy/elbv2/target_group_manager.go

```diff
@@ -233,7 +233,12 @@ func buildSDKModifyTargetGroupInput(tgSpec elbv2model.TargetGroupSpec) *elbv2sdk
 	return sdkObj
 }
 
-func buildSDKMatcher(modelMatcher elbv2model.HealthCheckMatcher) *elbv2sdk.Matcher {
+func buildSDKMatcher(modelMatcher elbv2model.HealthCheckMatcher, protocolVersion *elbv2model.ProtocolVersion) *elbv2sdk.Matcher {
```

Comment:
i think this is incorrect.
we should use modelMatcher, and check whether `modelMatcher.GPRC` and `modelMatcher.HTTPCode` is empty or not. (ideally we should change GRPC/HTTPCode to a string pointer as well, but keep it as string works as well)

Reply:
yeah, I was going between that option and this one myself. Makes sense.